Adds rod suplexing ability to legendary athletics

Bombermansam opened the pull request:
## About The Pull Request
Adds the ability for people with legendary physical ability to suplex rods with pure strength. Also adds missing level up text for every level above Novice
## Why It's Good For The Game
Every other skill has level up text except for athletics. This fixes it

The addition of the rod suplex ability to legendary athletics makes sense. RDs can suplex rods by using math, logic and proper application of strength (RDs are strong but not herculean in strength). This implies there is a way to suplex rods through pure strength alone, hence the change.

This also opens up the ability to get the rod stopper achievement to other people besides the RD. Some may say that this takes away from the RD's unique skill chip but it requires someone to be legendary athletics to perform this feat of strength. It isn't particularly difficult to get legendary athletics but it does require a significant time investment so I dont expect many people to grind for such a niche ability. Besides, it could offer up some interesting RP opportunities between Slab Bulkhead and Dr. Egghead to see who gets to suplex the rod.

Here's proof of testing if you need it. It compiles, it was tested locally, the level up text functions and the suplexing feature functions.

![image](https://github.com/user-attachments/assets/994521f2-63f3-47f8-aafe-2f66833a83d7)
![dreamseeker_W9i40O8lXg](https://github.com/user-attachments/assets/740b69e9-437e-41e6-ae90-f32be54a97a0)
## Changelog
:cl:
add: Added custom level up text for athletics like other skills
add: Added the ability to suplex rods for legendary athletics
/:cl:

Bombermansam committed:
Adds rod suplexing ability to legendary athletics

ElGitificador commented:
god bless i had to learn sci just to get that achievement (and still dont got it)

Jacquerel commented:
I don't know about this, I usually object to increasing access to special or rare things just on general principles that having fewer sources is more exciting.

moocowswag commented:
Wasnt the suplex put into rd skillchip or it still requires literally being the rd.

Bombermansam commented:
> I don't know about this, I usually object to increasing access to special or rare things just on general principles that having fewer sources is more exciting.

The suplex is an ability granted to the RD with the R.D.S.P.L.X skillchip. It says that it gives the RD the knowledge of leverage to stop a rod using the minimum possible force (paraphrasing). It makes sense to me that if the RD can stop a rod using the principles of leverage (knowledge and strength) then a person that has reached herculean strength would be able to stop the rod through pure physicality.

While I agree that making something less rare is generally a bad thing to do and I would wholly agree with you in most cases, I believe that this PR could be an exception. 

It takes a lot of tedious grinding just to get to legendary athletics in the first place. I do not expect many people to devote maybe a third to half of their round grinding this just to get a niche mechanical skill (besides the first week or so if this pr gets merged because "new thing"). I also believe that if the RD gets angry at Chad Lunkhead for suplexing a rod or vis versa this could provoke some conflict between the two providing some RP to the Rod event which is purely mechanical.

san7890 commented:
How long (in minutes) does it take to get legendary athletics (on average)?

Bombermansam commented:
> How long (in minutes) does it take to get legendary athletics (on average)?

I tested it experimentally 3 times in the time between now and your comment. It took 48 minutes, 53 minutes and 52 minutes for me to reach legendary athletics with an average of 51 minutes to reach this skill level. I tested it in normal conditions (no using the gravity generator or chems) with ample food nearby to reduce the time eating. I used a bench press machine for each experiment using the "work out" button on cooldown. Sleeping was done whenever the "Your muscles are exhausted! Might be a good idea to sleep..." message appeared to replicate actual conditions when leveling athletics. The experiments were done on a local host. I will not be testing the time it takes to reach legendary athletics again experimentally, this was extremely boring.

Kiromaka commented:
50 minutes seems too long, did you use sleep buffs ? sleep mask, headphones, pillow, blanky, bed ?

san7890 commented:
thank you for doing it manually, i wish you had reached out and we could have made a dummy mob or equation-based simulation to figure this out. i'm wondering if we shouldn't introduce a hard ceiling to where you can't get legendary athletics until a random duration of 55-60 minutes after the round elapsed (or figure out some algorithm that would place players to get this around that time).

The reason why I mention this is because the "sweet spot" that we want rounds to last is about 90 minutes, so I think that if you REALLY want to suplex rods then doing other things while grinding it out until 2/3rds of the round elapsed so you can deal with late-round station chaos should rods show up or enjoy those other benefits while antagonists are getting stronger and stronger...

I don't know if this PR is the best place to introduce the limit change, it's just an ideal. I think the present experimental range of 47-53 minutes of pure grinding is fine, but I feel like rounding it up to a more exact number would be more ideal in the future as it doesn't really infringe on the RD given that 3/5ths of the round and any early rods would be their exclusive domain. i'll ponder on this for a day or so in case anyone has any additional concerns they want to mention in this PR, but I'm learning towards merging @Jacquerel

Bombermansam commented:
> 50 minutes seems too long, did you use sleep buffs ? sleep mask, headphones, pillow, blanky, bed ?

I slept on the bench when testing it. This experiment was conducted to determine the baseline time it takes so I did not use anything other than the bare minimum to achieve the maximum skill. If sleep aids, stimulants, getting the genetics "Strong gene" and exercising within the gravity generator were used, its conceivable that it will lower the time to achieve max skill level by ~15 minutes from reduction in sleep time, reduction in stamina depletion, and the experience bonuses you get if you have the strength gene and under heavy gravity. You will still have to acquire these items and get the requisite access if you want to completely min/max athletics leveling which I am guessing will not take more than 5-10 minutes for an experienced greytider but it is something to consider. In a normal game someone would probably use the sleep aids at the very minimum which will reduce the amount of time it will take to level but I cannot empirically state the amount of time will be saved through this method. 



> thank you for doing it manually, i wish you had reached out and we could have made a dummy mob or equation-based simulation to figure this out. i'm wondering if we shouldn't introduce a hard ceiling to where you can't get legendary athletics until a random duration of 55-60 minutes after the round elapsed (or figure out some algorithm that would place players to get this around that time).
> 
> The reason why I mention this is because the "sweet spot" that we want rounds to last is about 90 minutes, so I think that if you REALLY want to suplex rods then doing other things while grinding it out until 2/3rds of the round elapsed so you can deal with late-round station chaos should rods show up or enjoy those other benefits while antagonists are getting stronger and stronger...
> 
> I don't know if this PR is the best place to introduce the limit change, it's just an ideal. I think the present experimental range of 47-53 minutes of pure grinding is fine, but I feel like rounding it up to a more exact number would be more ideal in the future as it doesn't really infringe on the RD given that 3/5ths of the round and any early rods would be their exclusive domain. i'll ponder on this for a day or so in case anyone has any additional concerns they want to mention in this PR, but I'm learning towards merging @Jacquerel

I tested the times under ideal circumstances but without aids to determine a baseline time, however due to the concerns of the previous commenter on the use of aids to reduce the training time I can perform the experiment again. If you require this data to make a decision I can repeat the experimental tests using aids to determine how much time it will save a prospective strongman. 

We could use a dummy mob or simulation to calculate this but finding this data experimentally will also help determine human variability in the times (and how boring it is to level). 

Let me know if you need me to test it again with the aids if you need the data to make a decision (because I'd rather not test it again if I dont have to. Haha).

Kiromaka commented:
It should take around 20 minutes with sleeping buffs + eggs as food.

Bombermansam commented:
> It should take around 20 minutes with sleeping buffs + eggs as food.

Thank you for testing the time with sleep aids. I just tested it with sleep aids and donk pockets as food and got 23, 25 and 25 minutes. I did not count the time it took to get the sleep aids because it is simple to obtain. I dont know how you got 20 minutes but I'll count that as skill. If we average all four data points then sleeping aids reduce the time it takes to get to legendary athletics by half compared to just normally exercising. So a person using the most optimum strategy while leveling athletics could reach legendary athletics in just under a third of a normal 90 minute round.

san7890 commented:
20-25 minutes being accomplishable is way too short for this. Maybe make a new "mythical" tier which doesn't require that much effort but is timegated to a round of in game time. Otherwise, this PR is just simply not mergeable IMO since it could really optimize out some of the only special things the RD has (beyond turtlenecks)

Bombermansam commented:
> 20-25 minutes being accomplishable is way too short for this. Maybe make a new "mythical" tier which doesn't require that much effort but is timegated to a round of in game time. Otherwise, this PR is just simply not mergeable IMO since it could really optimize out some of the only special things the RD has (beyond turtlenecks)

That's understandable, let me see what I could do to make this more in line with your recommendations.

Kiromaka commented:
> That's understandable, let me see what I could do to make this more in line with your recommendations.

Dont time gate the achievement tho.

Bombermansam committed:
Adds Mythical Skill tier. Moves Athletics Rod Suplex to Mythical Tier.

Bombermansam committed:
Fixes a couple commas

Bombermansam committed:
Adds mythical to code skill subsystem

Bombermansam committed:
Fixes Admin Skill Panel for new Skill Level

Bombermansam committed:
Fixes the lists so I dont get the list index out of bounds error

Bombermansam committed:
Same as before but for real this time

Bombermansam committed:
Fixes the skill bar in the admin skill panel

Bombermansam committed:
Gives the remaining skills appropriate buffs for the mythical skill level

Bombermansam commented:
> 20-25 minutes being accomplishable is way too short for this. Maybe make a new "mythical" tier which doesn't require that much effort but is timegated to a round of in game time. Otherwise, this PR is just simply not mergeable IMO since it could really optimize out some of the only special things the RD has (beyond turtlenecks)

I added the Mythical Tier of skill level as per your request. This level requires 4000 more experience than the previous level and it is quite an feat if you get it. It should take an extra 30 minutes to get to this level from legendary. If you use the previous data then it should take between 45-50 minutes to obtain mythical skill level in athletics(using the experimental procedure we had developed). Suplexing was moved to be the ultimate reward for the athletics skill track.

![image](https://github.com/user-attachments/assets/8cd573f4-8987-406f-9638-6396dec655c3)

Adding the Mythical Skill level in the way that I did also adds the mythical skill level for the other skills (cleaning, mining, gaming and fishing). I fixed the lists in these skills to correctly represent the 8 skill levels instead of 7 and gave them appropriate skill modifiers for this level. For example cleaning speed going from 0.36 for Legendary skill to 0.25 for Mythical skill. There are no physicalized rewards beyond the skill boosts so this does create a reward gap in comparison to the athletics one. This can be solved in another PR because this one is already straining the limits of its scope.

SmArtKar commented:
I'm not a fan of this, rod suplexing is an RD thing and should probably remain as such.
